### ISSUE TITLE
na.strings in fread 

### ISSUE THREAD

**Issue by msgoussi** [Rdatatable/data.table#2927]
When I use fread, I get this error (NAstring <<0>> is recognized as type boolean, this is not permitted)
na.strings = c("",        "-",       "_",       "..",      "...",     "--",      "***",     "*" ,
"n/a",     "n.a.",    "#VALUE!", "0",       "Inf",     "-Inf",    "NAN", "r", "e")
  
If I removed "0" from na.strings, fread is not getting error.

However, cells that contains "r" or "e", is not converted to NA, and their columns are characters.

Please advise.

**Comment by MichaelChirico** [comment]
Please include your `data.table` package version & system info, and the file itself, if you can.

**Comment by HughParsonage** [comment]
``` r
fread("a,b,c\n0,1,2", na.strings = "0")
#> Error in fread("a,b,c\n0,1,2", na.strings = "0"): could not find function "fread"
```

Created on 2018-06-11 by the [reprex package](http://reprex.tidyverse.org) (v0.2.0).

**Comment by HughParsonage** [comment]
``` r
data.table::fread("a,b,c\n0,1,2", na.strings = "0", verbose = TRUE)
#> Input contains a \n or is "". Taking this to be text input (not a filename)
#> [01] Check arguments
#>   Using 12 threads (omp_get_max_threads()=12, nth=12)
#> Error in data.table::fread("a,b,c\n0,1,2", na.strings = "0", verbose = TRUE):
#>   freadMain: NAstring <<0>> is recognized as type boolean, this is not permitted.
```

**Comment by msgoussi** [comment]
Sys.info()
          sysname           release           version          nodename 
        "Windows"        ">= 8 x64"      "build 9200" "IRT-310677-Z440" 
          machine             login              user    effective_user 
         "x86-64"          "310677"          "310677"          "310677" 

data.table package: 1.11.4

**Comment by MichaelChirico** [comment]
Thank @HughParsonage, darn that looks no bueno.

I guess there should be some interaction with `logical01` but there doesn't appear to be.

**Comment by msgoussi** [comment]
Is it a bug?

**Comment by jangorecki** [comment]
@msgoussi yes, you can workaround it with something like
```r
data.table::fread("a,b,c\n0,1,2")[a==0, a:=NA]
```

**Comment by msgoussi** [comment]
If i have 500 columns and i need to clean colmuns and consider the following strings , c("", "-", "_", "..", "...", "--", "**", "" ,
"n/a", "n.a.", "#VALUE!", "0", "Inf", "-Inf", "NAN", "r", "e"), as na. The way around will not look good. This is my opnion

**Comment by MichaelChirico** [comment]
you're correct, but until the bug is fixed, you can at least get rid of all
the other NA strings with fread, then only replace 0 in the code.

**Comment by ribailey** [comment]
I have an R package on Github, ribailey/gghybrid, which includes a function to read genomic data files with potentially millions of columns, and one of the main softwares for producing these input files, PLINK, always codes missing data as zero. I use fread within my function to read in the data and declare missing values. This means I can't read in the most common file type people might want to use, due to the bug described here. Has there been any progress on this? Many thanks, Richard.

**Comment by ben-schwen** [comment]
Removing `|| strcmp(ch,"1")==0 || strcmp(ch,"0")==0` from `fread.c` seems like an option to "fix" this.

Only breaks 1 test case which is explicitly testing for na.strings = '1'.

Interaction with `logical01` seems also legit.

```
data.table::fread("a,b,c\n0,1,2\n1,0,2", na.strings = "0", logical01=T)

        a      b     c                                                                                                     
    <lgcl> <lgcl> <int>                                                                                                  
1:     NA   TRUE     2                                                                                                  
2:   TRUE     NA     2 
```

**Comment by shrektan** [comment]
I think the expected behavior is that we should not allow `logical01=TRUE` and `na.strings = "0"` at the same time. If this is agreed, I'm happy to start to file a PR for this.

**Comment by MichaelChirico** [comment]
Agreed @shrektan. Nor `na.strings = "1"` though I guess that's a pretty obscure use case.
